isis/src/base/apps/getsn/main.cpp

```diff
@@ -1,93 +1,40 @@
 #include "Isis.h"
 
-#include <iostream>
-#include <sstream>
-#include <QString>
+#include "getsn.h"
+#include "PvlGroup.h"
+#include "Pvl.h"
 
 #include "Blob.h"
 #include "Cube.h"
 #include "History.h"
 #include "ObservationNumber.h"
 #include "Preference.h"
-#include "Pvl.h"
+
 #include "SerialNumber.h"
 #include "SessionLog.h"
 #include "TextFile.h"
 
+
 using namespace Isis;
 using namespace std;
 
 void IsisMain() {
+
   // Set Preferences to always turn off Terminal Output
   PvlGroup &grp = Isis::Preference::Preferences().findGroup("SessionLog", Isis::Pvl::Traverse);
   grp["TerminalOutput"].setValue("Off");
 
-  // Open the input cube
   UserInterface &ui = Application::GetUserInterface();
-  Cube cube;
-  QString from = ui.GetFileName("FROM");
-  cube.open(from, "r");
-
-  // Determine if output should be written base on parameters
-  bool WriteFile = ui.GetBoolean("FILE");
-  bool WriteSN = ui.GetBoolean("SN");
-  bool WriteObservation = ui.GetBoolean("OBSERVATION");
-
-  QString format = ui.GetString("FORMAT");
-  bool pvl;
-  if (format == "PVL") {
-    pvl = true;
-  }
-  else {
-    pvl = false;
-  }
-
-  // Extract label from cube file
-  Pvl *label = cube.label();
-
-  PvlGroup sn("Results");
+  Pvl appLog;
 
-  if (WriteFile) sn += PvlKeyword("Filename", from);
-  if (WriteSN) sn += PvlKeyword( "SerialNumber", SerialNumber::Compose( *label, ui.GetBoolean("DEFAULT") ) );
-  if (WriteObservation) sn += PvlKeyword( "ObservationNumber", ObservationNumber::Compose( *label, ui.GetBoolean("DEFAULT") ) );
+  getsn(ui, &appLog);
```

Comment:
You need to wrap this in a try catch block so that it properly logs even if the function errors. See https://github.com/USGS-Astrogeology/ISIS3/blob/d9049c1941301837fce014ea59f046af1c69c6a6/isis/src/base/apps/spiceinit/main.cpp#L15

Reply:
If the function errors, the exception that is thrown naturally is still logged, is swallowing the error an re-interpreting it a good idea?

Reply:
This caused an issue in spiceinit where the exception didn't report what kernel sets were attempted prior to failure, that was done via the log. In this case, I think the exception will be sufficient.